corva/models/stream.py

```diff
@@ -71,16 +74,55 @@ def app_stream_id(self) -> int:
         return self.metadata.app_stream_id
 
     @property
-    def is_completed(self) -> bool:
+    def _is_completed(self) -> bool:
         """there can only be 1 completed record, always located at the end of the list"""
 
         if self.records:
             return self.records[-1].collection == 'wits.completed'
 
         return False
 
+    @pydantic.root_validator(pre=False, skip_on_failure=True)
+    def require_app_key_in_metadata_apps(cls, values):
+        if values['app_key'] not in values['metadata'].apps:
```

Comment:
@ville is this right, or do we pass App ID integers to the apps array?

Reply:
```
    "metadata": {
      "apps": {
        "my-company.my-app": {
          "app_connection_id": 123
        },
        "big-data-energy.some-other-app": {
          "app_connection_id": 455
        }
        ...
```

Reply:
thanks, resolving